web/src/edit/NeighbourhoodMode.svelte

```diff
@@ -332,126 +330,157 @@
+            alt="Restrict turns"
           />
-        </div>
-      </button>
-      <button
-        on:click={() => (action = startTurnRestrictionAction())}
-        disabled={action.kind == "turn_restriction"}
-        class:active={action.kind == "turn_restriction"}
-        class:outline={action.kind != "turn_restriction"}
-        data-tooltip="Restrict turns (hotkey 4)"
-      >
-        <img
-          src={`${import.meta.env.BASE_URL}/filters/no_right_turn.svg`}
-          alt="Restrict turns"
-        />
-      </button>
-      <button
-        on:click={() => (action = { kind: "main-roads" })}
-        disabled={action.kind == "main-roads"}
-        class:active={action.kind == "main-roads"}
-        class:outline={action.kind != "main-roads"}
-        data-tooltip="Reclassify main roads (hotkey 5)"
-      >
-        <img src={mainRoadIconUrl} alt="Change main/minor roads" />
-      </button>
+        </button>
+        <button
+          on:click={() => (action = { kind: "main-roads" })}
+          disabled={action.kind == "main-roads"}
+          class="icon-btn"
+          class:active={action.kind == "main-roads"}
+          class:outline={action.kind != "main-roads"}
+          data-tooltip="Reclassify main roads (hotkey 5)"
+        >
+          <img src={mainRoadIconUrl} alt="Change main/minor roads" />
+        </button>
+      </div>
+      <div style="display: flex; justify-content: right; gap: 8px;">
+        <button
+          class="outline icon-btn"
+          disabled={undoLength == 0}
+          on:click={undo}
+          data-tooltip={undoLength == 0
+            ? "Undo Ctrl+Z"
+            : `Undo (${undoLength}) Ctrl+Z`}
+        >
+          <Undo />
+        </button>
+        <button
+          class="outline icon-btn"
+          disabled={redoLength == 0}
+          on:click={redo}
+          data-tooltip={redoLength == 0
+            ? "Redo Ctrl+Y"
+            : `Redo (${redoLength}) Ctrl+Y`}
+        >
+          <Redo />
+        </button>
+      </div>
     </div>
 
-    <button class="outline" on:click={() => (settingFilterType = true)}>
-      Change modal filter type
-    </button>
-    <ChangeFilterModal bind:show={settingFilterType} />
-
-    <div style="display: flex; justify-content: space-between;">
-      <button
-        class="outline"
-        disabled={undoLength == 0}
-        on:click={undo}
-        data-tooltip="Ctrl+Z"
-      >
-        {#if undoLength == 0}
-          Undo
-        {:else}
-          Undo ({undoLength})
-        {/if}
-      </button>
-      <button
-        class="outline"
-        disabled={redoLength == 0}
-        on:click={redo}
-        data-tooltip="Ctrl+Y"
-      >
-        {#if redoLength == 0}
-          Redo
-        {:else}
-          Redo ({redoLength})
-        {/if}
-      </button>
+    <div
+      style="min-height: 200px; padding-bottom: 16px; border-bottom: solid var(--pico-muted-border-color) 1px;"
```

Comment:
Oops, move that comment above down here. Maybe the min height needs to be a little higher. I'm guessing you used a browser inspector to measure the height of all cases, then picked the max for this value?